### PR TITLE
AC-6662: 500 error received when attempting to delete a user

### PR DESCRIPTION
**Pull request by VincentHokie**
#### Changes introduced in [AC-6662](https://masschallenge.atlassian.net/browse/AC-6662)
- stop logic from auto creating profiles on query set `get`, instead do that on user creation

#### How to test
- with a clean db
- while on development on both accelerate and django-accelerator
- log in as demoadmin
- try and delete [30-user](http://localhost:8181/admin/simpleuser/user/30/delete/) and [38298-user](http://localhost:8181/admin/simpleuser/user/38298/delete/), notice that both of these fail.
    - 38298-user has PartnerTeamMembers associated
    - 30-user has both PartnerTeamMembers and StartupStatus associated
- now checkout AC-6662 on both accelerate and django-accelerator
- try and delete [30-user](http://localhost:8181/admin/simpleuser/user/30/delete/) and [38298-user](http://localhost:8181/admin/simpleuser/user/38298/delete/) again, notice that it now passes. Success🏆 

#### Note
This PR has a sibling on [accelerate](https://github.com/masschallenge/accelerate/pull/2154) and this should be merged in first

### CODE REVIEW

**Comment by pbx** [comment]
Taking QA.

File: accelerator/models/profile_query_set.py [hunk 1]
```diff
@@ -56,22 +53,13 @@ def get(self, *args, **kwargs):
...
-            try:
-                self.user = User.objects.get(email=kwargs[EMAIL_KEY])
-            except User.MultipleObjectsReturned:
-                # This guard is necessary since emails are not necessarily
-                # unique in our system.
-                # Should be able to remove this when we define a custom User
-                self.user = User.objects.filter(
-                    email=kwargs[EMAIL_KEY]).first()
+            self.user = User.objects.get(email=kwargs[EMAIL_KEY])
+
```
Comment (pbx) [inline accelerator/models/profile_query_set.py]: Hurrah for deleting old crufty code

File: settings.py [hunk 2]
```diff
@@ -2,6 +2,12 @@
 # Copyright (c) 2017 MassChallenge, Inc.
 
 import os
+import sys
+import logging
+
+if len(sys.argv) > 1 and sys.argv[1] == 'test':
+    logging.disable(logging.CRITICAL)
```
Comment (pbx) [inline settings.py]: Why did you add this? Generally `if test` is something we want to avoid.

**Comment by VincentHokie** [comment]
Will continue to address concerns after @jonkiparsky 's concerns are discussed today.

File: settings.py (hunk 2 above)
Comment (VincentHokie) [reply settings.py]: logging was [suggested](https://github.com/masschallenge/django-accelerator/pull/174#discussion_r277315271) when a persons Member profile is deleted in favour of an Expert/ Entrepreneur profile.

Without this logic, our tests are flooded with those log messages.

**Comment by VincentHokie** [comment]
Closing this PR pending a discussion by team around this use case
